demo/admin/src/products/future/generated/ProductsGrid.tsx

```diff
@@ -99,7 +101,34 @@ export function ProductsGrid({ filter, toolbarAction, rowAction }: Props): React
     const dataGridProps = { ...useDataGridRemote(), ...usePersistentColumnState("ProductsGrid") };
 
     const columns: GridColDef<GQLProductsGridFutureFragment>[] = [
-        { field: "inStock", headerName: intl.formatMessage({ id: "product.inStock", defaultMessage: "In stock" }), type: "boolean", width: 90 },
+        {
+            field: "inStock",
+            headerName: intl.formatMessage({ id: "product.inStock", defaultMessage: "In Stock" }),
+            type: "singleSelect",
+            valueOptions: [
+                { value: "true", label: intl.formatMessage({ id: "product.inStock.true.primary", defaultMessage: "In stock" }) },
+                { value: "false", label: intl.formatMessage({ id: "product.inStock.false.primary", defaultMessage: "Out of stock" }) },
+            ],
+            renderCell: ({ row }) => {
+                const valueLabels: Record<string, React.ReactNode> = {
+                    true: (
+                        <GridCellContent
+                            primaryText={intl.formatMessage({ id: "product.inStock.true.primary", defaultMessage: "In stock" })}
+                            icon={<StateFilledIcon color="success" />}
+                        />
+                    ),
+                    false: (
+                        <GridCellContent
+                            primaryText={intl.formatMessage({ id: "product.inStock.false.primary", defaultMessage: "Out of stock" })}
+                            icon={<StateFilledIcon color="error" />}
+                        />
+                    ),
+                };
+                return row.inStock.toString() in valueLabels ? valueLabels[row.inStock.toString()] : row.inStock.toString();
```

Comment:
Is this fallback necessary?

Reply:
It adds some safety in case some value stored in the database does not exist in the `values` array/object or in the code in general 🤷🏼‍♂️